src/plans/D3MMIP21Plan.sol

```diff
@@ -0,0 +1,156 @@
+    function balanceOf(address) external view returns (uint256);
+}
+
+/// Combines the MIP21 Input and Output conduits
+contract D3MMIP21Plan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    mapping (address => uint256) public can;
+    mapping (address => uint256) public bud;
+    address                      public to;
+    uint256                      public cap;
+
+    TokenLike public immutable dai;
+    TokenLike public immutable gov;
+    address public immutable escrow;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event Hope(address indexed usr);
+    event Nope(address indexed usr);
+    event Kiss(address indexed who);
+    event Diss(address indexed who);
+    event Pick(address indexed who);
+    event Push(address indexed to, uint256 wad);
+
+    constructor(address dai_, address gov_, address escrow_) {
+        dai = TokenLike(dai_);
+        gov = TokenLike(gov_);
+        escrow = escrow_;
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MMIP21Plan/not-authorized");
+        _;
+    }
+
+    modifier operator {
+        require(can[msg.sender] == 1, "D3MMIP21Plan/not-operator");
+        _;
+    }
+
+    // --- Math ---
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function hope(address usr) external auth {
+        can[usr] = 1;
+        emit Hope(usr);
+    }
+    function nope(address usr) external auth {
+        can[usr] = 0;
+        emit Nope(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "cap") {
+            cap = data;
+        } else revert("D3MMIP21Plan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function kiss(address who) public auth {
+        bud[who] = 1;
+        emit Kiss(who);
+    }
+    function diss(address who) public auth {
+        if (to == who) to = address(0);
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    // --- Plan Interface ---
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        currentAssets;
+        return cap;
+    }
+
+    function active() public pure override returns (bool) {
+        return true;
+    }
+
+    function disable() external override {
+        require(
+            wards[msg.sender] == 1 ||
+            !active()
+        , "D3MMIP21Plan/not-authorized");
+        cap = 0;
+        emit Disable();
+    }
+
+    // --- Routing ---
```

Comment:
I think these routing functions should be on the pool, since that maintains balances rather than wind/unwind logic.  We may even just leave this as conduits, and add onto the conduits the functions the plan or pool need to function.

Reply:
Going to explain my mental model of Plan vs Pool with some pretty pictures. TLDR is pool should be dumb and stateless. Plan should have all logic / state.